applications/FluidDynamicsApplication/custom_constitutive/newtonian_nodal_properties_3d_law.h

```diff
@@ -0,0 +1,222 @@
+//
+//  Main authors:    Daniel Diez
+//
+
+#if !defined (KRATOS_NEWTONIAN_LAW_NODAL_PROPERTIES_3D_H_INCLUDED)
+#define  KRATOS_NEWTONIAN_LAW_NODAL_PROPERTIES_3D_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "newtonian_3d_law.h"
+#include "custom_utilities/element_size_calculator.h"
+#include "includes/cfd_variables.h"
+
+
+namespace Kratos
+{
+
+/**
+ * Defines a Newtonian constitutive law in 3D.
+ * This material law is defined by the parameters:
+ * 1) DYNAMIC_VISCOSITY (read from the nodes!!)
+ * 2) C_SMAGORINSKY
+ */
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) NewtonianNodalProperties3DLaw : public Newtonian3DLaw
+{
+public:
+    /**
+     * Type Definitions
+     */
+    typedef std::size_t             SizeType;
+    
+    /**
+     * Counted pointer of NewtonianNodalProperties3DLaw
+     */
+
+    KRATOS_CLASS_POINTER_DEFINITION(NewtonianNodalProperties3DLaw);
+
+    /**
+     * Life Cycle
+     */
+
+    /**
+     * Default constructor.
+     */
+    NewtonianNodalProperties3DLaw()
+        : Newtonian3DLaw()  {}
+
+    /**
+     * Clone function (has to be implemented by any derived class)
+     * @return a pointer to a new instance of this constitutive law
+     */
+    ConstitutiveLaw::Pointer Clone() const override {
+        return Kratos::make_shared<NewtonianNodalProperties3DLaw>(*this);
+    }
+
+    /**
+     * Copy constructor.
+     */
+    NewtonianNodalProperties3DLaw (const NewtonianNodalProperties3DLaw& rOther)
+        : Newtonian3DLaw()  {}
+
+
+    /**
+     * Destructor.
+     */
+    ~NewtonianNodalProperties3DLaw() override {}
+
+
+    /**
+     * Input and output
+     */
+
+    /**
+     * Turn back information as a string.
+     */
+    std::string Info() const override {
+        return "NewtonianNodalProperties3DLaw";
+    }
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+    ///@}
+    ///@name Protected Operators
+    ///@{
+    ///@}
+    ///@name Protected Operations
+    ///@{
+    
+    
+    double ComputeEffectiveViscosity(ConstitutiveLaw::Parameters& rParameters) const override {
+
+        double viscosity;
+        EvaluateInPoint(viscosity, DYNAMIC_VISCOSITY, rParameters);
+        
+        const Properties& prop = rParameters.GetMaterialProperties();
+
+        if (prop.Has(C_SMAGORINSKY)) {
+            const double csmag = prop[C_SMAGORINSKY];
+            if (csmag > 0.0) {
+                const double strain_rate = EquivalentStrainRate(rParameters);
+                const BoundedMatrix<double, 4, 3>& DN_DX = rParameters.GetShapeFunctionsDerivatives();
+                const double elem_size = ElementSizeCalculator<3,4>::GradientsElementSize(DN_DX);
+                double length_scale = csmag * elem_size;
+                length_scale *= length_scale;
+                viscosity += 2.0*length_scale * strain_rate;
```

Comment:
Be careful here, the Smagorinsky model gives you a kinematic viscosity. If `C_SMAGORINSKY` is a dimensionless number, you need to multiply times the density if you want to add it to a dynamic viscosity.

Reply:
You are right! Fixed